cmake/templates/BoostConfig.cmake.in

```diff
@@ -102,3 +102,11 @@ if(TARGET Boost::iostreams)
   include("${CMAKE_CURRENT_LIST_DIR}/BoostZLIBDependency.cmake" OPTIONAL)
   include("${CMAKE_CURRENT_LIST_DIR}/BoostBZip2Dependency.cmake" OPTIONAL)
 endif()
+
+if(TARGET Boost::thread)
+  find_package(Threads REQUIRED)
+  set_property(
+    TARGET "Boost::thread"
+    APPEND PROPERTY INTERFACE_LINK_LIBRARIES Threads::Threads
```

Comment:
Error on Linux reproduced now:
* https://s3.amazonaws.com/archive.travis-ci.org/jobs/157663063/log.txt
```
thread.cpp:(.text+0x2c5): undefined reference to `clock_gettime'
```

Can be fixed by adding `-lrt`: http://stackoverflow.com/a/2418175/2288008
In CMake terms this will be:
```cmake
include(CheckLibraryExists) # check_library_exists
check_library_exists(rt clock_gettime "" _boost_check_rt_library)
if(_boost_check_rt_library)
  # Add library to 'INTERFACE_LINK_LIBRARIES' of Boost::thread target
endif()
```